eurocodedesign/units/__init__.py

```diff
@@ -0,0 +1,255 @@
+from abc import ABC
+from enum import Enum, unique, auto
+from functools import partial
+from typing import Self, TypeAlias
+
+
+@unique
+class PhysicalType(Enum):
+    DIMENSIONLESS = auto()
+    ANGLE = auto()
+    TIME = auto()
+    LENGTH = auto()
+    AREA = auto()
+    VOLUME = auto()
+    SECOND_MOMENT_OF_AREA = auto()
+    SPEED = auto()
+    ACCELERATION = auto()
+    MASS = auto()
+    FORCE = auto()
+    PRESSURE = auto()
+    ENERGY = auto()
+    TEMPERATURE = auto()
+
+
+class Prefix(Enum):
+    "Metric prefixes from nano (1e-9) to giga (1e9)"
+    giga = G = 1e9
+    mega = M = 1e6
+    kilo = k = 1e3
+    hecto = h = 1e2
+    deca = da = 1e1
+    none = one = 1e0
+    deci = d = 1e-1
+    centi = c = 1e-2
+    milli = m = 1e-3
+    micro = µ = 1e-6
+    nano = n = 1e-9
+
+
+_allowed_multiplications = {}
```

Comment:
I think this can be deleted? allowed multiplications is also defined in line 232

Reply:
I think it is necessary, because otherwise there will be an compilation error, because I refer to the variable in AbstractUnit before it is declared. However, the specific unit types can only be put into the dict, after their definition